lib/elixir/pages/Operators.md

```diff
@@ -0,0 +1,105 @@
+<code>&#124;&#124;</code> <code>&#124;&#124;&#124;</code> `or`                           | Left to right
+`=`                                                                                      | Right to left
+`=>`                                                                                     | Right to left
+<code>&#124;</code>                                                                      | Right to left
+`::`                                                                                     | Right to left
+`when`                                                                                   | Right to left
+`<-`, `\\`                                                                               | Left to right
+`&`                                                                                      | Unary
+
+## Defining custom operators
+
+Elixir is capable of parsing a predefined set of operators; this means that it's not possible to define new operators (like one could do in Haskell, for example). However, not all operators that Elixir can parse are *used* by Elixir: for example, `+` and `||` are used by Elixir for addition and boolean *or*, but `<~>` is not used (but valid).
+
+To define an operator, you can use the usual `def*` constructs (`def`, `defp`, `defmacro`, and so on) but with a syntax similar to how the operator is used:
+
+```elixir
+defmodule MyOperators do
+  # We define ~> to return the maximum of the given two numbers,
+  # and <~ to return the minimum.
+
+  def a ~> b, do: max(a, b)
+  def a <~ b, do: min(a, b)
+end
+```
+
+To use the newly defined operators, we **have to** import the module that defines them:
+
+```elixir
+iex> import MyOperators
+iex> 1 ~> 2
+2
+iex> 1 <~ 2
+1
+```
+
+The following is a table of all the operators that Elixir is capable of parsing, but that are not used by default:
+
+* `|`
+* `|||`
+* `&&&`
+* `<<<`
+* `>>>`
+* `~>>`
+* `<<~`
+* `~>`
+* `<~`
+* `<~>`
+* `<|>`
+* `^^^`
+* `~~~`
+
+The following operators are used by the `Bitwise` module when imported: `&&&`, `^^^`, `<<<`, `>>>`, `|||`, `~~~`. See the documentation for `Bitwise` for more information.
+
+## Redefining existing operators
+
+The operators that Elixir uses (for example, `+`) can be defined by any module and used in place of the ones defined by Elixir, provided they're specifically not imported from `Kernel` (which is imported everywhere by default). For example:
+
+```elixir
+defmodule NastyMath do
+  # Let's be nasty by changing the meaning of +:
+  def a + b, do: a - b
+end
+```
+
+Now, we will get an error if we try to use this operator "out of the box":
+
+```elixir
+iex> import NastyMath
+iex> 1 + 2
+** (CompileError) iex:11: function +/2 imported from both NastyMath and Kernel, call is ambiguous
+```
+
+So, as mentioned above, we need to explicitly *not* import `+/2` from `Kernel`:
+
+```elixir
+iex> import NastyMath
```

Comment:
Ahah, I used "nasty" because it's not just bad, it's *evil* to swap + for - 😄 I'll use `BadMath`, no problem!